homeassistant/components/integration/sensor.py

```diff
@@ -174,22 +174,24 @@ def _unit(self, source_unit: str) -> str:
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
         await super().async_added_to_hass()
-        if state := await self.async_get_last_state():
-            try:
-                self._state = Decimal(state.state)
-            except (DecimalException, ValueError) as err:
-                _LOGGER.warning(
-                    "%s could not restore last state %s: %s",
-                    self.entity_id,
-                    state.state,
-                    err,
-                )
+        if (state := await self.async_get_last_state()) is not None:
+            if state.state in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+                self._attr_available = False
```

Comment:
This seems odd, if the restored last state is `unknown`, the sensor becomes `unavailable`?

Reply:
My logic: if the restore state is Unknown there is no valid value available to show the user, and therefore it would be best to make the sensor unavailable. 

On a second look if the cast fails, I would also need to set the sensor unavailable.

What do you think ? (I'll change accordingly in either options)

Reply:
> My logic: if the restore state is Unknown there is no valid value available to show the user, and therefore it would be best to make the sensor unavailable.

That that make the assumption they are the same, which they are not.

> On a second look if the cast fails, I would also need to set the sensor unavailable.

Or unknown?

IMHO, unavailable only applies to the integration sensor when the source entity isn't available.

Reply:
I therefore think that UNKNOWN has no place in this integration:
- If the source issues an unknown, the implementation has always ignored the value, skipping it (sensor.py:210)
- If the state can't be UNKNOWN, I believe the result of "last_state" can't also be UNKNOWN (might be wrong on this one)

I've updated the PR accordingly